src/components/CalendarContainer.ts

```diff
@@ -0,0 +1,526 @@
+                error: error => window.mx.ui.error(
+                    `Error while executing microflow: ${onResizeMicroflow}: ${error.message}`
+                )
+            });
+        } else if (onResizeEvent === "callNanoflow" && onResizeNanoflow.nanoflow) {
+            window.mx.data.callNanoflow({
+                nanoflow: onResizeNanoflow,
+                origin: mxform,
+                context,
+                error: error => window.mx.ui.error(
+                    `An error occurred while executing the nanoflow: ${error.message}`
+                )
+            });
+        }
+    }
+
+    private onClickSlot = (slotInfo: any) => {
+        mx.data.create({
+            entity: this.props.eventEntity,
+            callback: (object) => {
+                object.set(this.props.titleAttribute, object.get(this.props.titleAttribute));
+                object.set(this.props.eventColor, object.get(this.props.titleAttribute));
+                object.set(this.props.startAttribute, slotInfo.start);
+                object.set(this.props.endAttribute, slotInfo.end);
+                this.excecuteSlotAction(object);
+            },
+            error: error => window.mx.ui.error(`Error while creating a new event: ${ error.message }`)
+        });
+    }
+
+    private excecuteSlotAction(mxObject: mendix.lib.MxObject) {
+        const { onClickSlotEvent, slotMicroflow, mxform, slotNanoflow } = this.props;
+        const context = new mendix.lib.MxContext();
+        context.setContext(mxObject.getEntity(), mxObject.getGuid());
+        if (onClickSlotEvent === "callMicroflow" && slotMicroflow && mxObject.getGuid()) {
+            window.mx.ui.action(slotMicroflow, {
+                context,
+                origin: mxform,
+                error: error => window.mx.ui.error(
+                    `Error while executing microflow: ${slotMicroflow}: ${error.message}`
+                )
+            });
+        } else if (onClickSlotEvent === "callNanoflow" && slotNanoflow.nanoflow) {
+            window.mx.data.callNanoflow({
+                nanoflow: slotNanoflow,
+                origin: mxform,
+                context,
+                error: error => window.mx.ui.error(
+                    `An error occurred while executing the nanoflow: ${error.message}`
+                )
+            });
+        }
+    }
+
+    private onDropEvent = (eventInfo: any) => {
+        const { events } = this.state;
+        const eventPosition = events.indexOf(eventInfo.event);
+        const updatedEvent: CalendarEvent = {
+            title: eventInfo.event.title,
+            allDay: eventInfo.event.allDay,
+            start: eventInfo.start,
+            end: eventInfo.end,
+            guid: eventInfo.event.guid,
+            color: eventInfo.event.color
+        };
+        const nextEvents = [ ...events ];
+        nextEvents.splice(eventPosition, 1, updatedEvent);
+        this.setState({ events: nextEvents });
+        mx.data.get({
```

Comment:
No need to get event again from the server.
Store reference in the widget.
Though for XPath use the new ales the objects are not cashed

``` ts
interface onlineData {
    getByXpath(xpath: string, options?:{ count: boolean, sort: Sort[] }): Promise<{count: number, mxobjs: mendix.lib.MxObject[]}>;
}
```